src/org/zaproxy/zap/extension/callback/OptionsCallbackPanel.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.callback;
+
+import java.awt.CardLayout;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.List;
+
+import javax.swing.JCheckBox;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.OptionsParam;
+import org.parosproxy.paros.view.AbstractParamPanel;
+import org.zaproxy.zap.utils.ZapPortNumberSpinner;
+import org.zaproxy.zap.utils.ZapTextField;
+import org.zaproxy.zap.view.LayoutHelper;
+
+public class OptionsCallbackPanel extends AbstractParamPanel {
+
+    private static final long serialVersionUID = 1L;
+    private JPanel panel = null;
+
+    private ExtensionCallback ext;
+    private JComboBox<String> localAddress = null;
+    private JComboBox<String> remoteAddress = null;
+    private ZapTextField testURL = null;
+    private JCheckBox randomPort = null;
+    private ZapPortNumberSpinner spinnerPort = null;
+
+    public OptionsCallbackPanel(ExtensionCallback ext) {
+        super();
+        this.ext = ext;
+        this.setLayout(new CardLayout());
+        this.setName(Constant.messages.getString("callback.options.title"));
+        this.add(getPanelLocalProxy(), getPanelLocalProxy().getName());
+
+    }
+
+    /**
+     * This method initializes panelLocalProxy
+     *
+     * @return javax.swing.JPanel
+     */
+    private JPanel getPanelLocalProxy() {

```

Comment:
Would it be more clear to call this callbackProxy or something like that....it isn't really "the" local zap proxy, if I'm following correctly.